Dockerfile

```diff
@@ -1,89 +1,45 @@
-    apt-get install -y python3-pip && \
-    apt-get install -y python3.9-venv
-
-RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.9 10
+    apt-get install -y python3.9 python3.9-dev python3-pip python3.9-venv && \
+    update-alternatives --install /usr/bin/python python /usr/bin/python3.9 10 && \
+    python -m venv mava && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
 
-# Check python v
-RUN python -V
-
-# Setup virtual env
-RUN python -m venv mava
+# Setup virtual env and path
 ENV VIRTUAL_ENV /mava
 ENV PATH /mava/bin:$PATH
-RUN pip install --upgrade pip setuptools wheel
+
 # Location of mava folder
 ARG folder=/home/app/mava
-## working directory
+
+# Set working directory
 WORKDIR ${folder}
-## Copy code from current path
-COPY . /home/app/mava
-# For box2d
-RUN apt-get install swig -y
-## Install core dependencies + reverb.
-RUN pip install -e .[reverb]
-## Optional install for screen recording.
-ENV DISPLAY=:0
-RUN if [ "$record" = "true" ]; then \
-    ./bash_scripts/install_record.sh; \
-    fi
-EXPOSE 6006
-##########################################################
 
-# Jax Images
-##########################################################
-# Core Mava image
-FROM mava-core as jax-core
-# Jax gpu config.
-ENV XLA_PYTHON_CLIENT_PREALLOCATE=false
-## Install core jax dependencies.
-# Install jax gpu
-RUN pip install -e .[jax]
-# TODO (Ruan): This version is pinned now to avoid a breaking change in jaxlib.
-# Unpin when the issue is resolved.
-# Please see https://github.com/deepmind/dm-haiku/issues/565 for details.
-RUN pip install --upgrade "jax[cuda]==0.3.24" -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html
-##########################################################
+# Copy all code needed to install dependencies
+COPY ./requirements ./requirements
+COPY setup.py .
+COPY README.md .
+COPY mava/version.py mava/version.py
 
-##########################################################
-# PZ image
-FROM jax-core AS pz
-RUN pip install -e .[pz]
-# PettingZoo Atari envs
-RUN apt-get update
-RUN apt-get install ffmpeg libsm6 libxext6  -y
-RUN apt-get install -y unrar-free
-RUN pip install autorom
-RUN AutoROM -v
-##########################################################
+RUN echo "Installing requirements..."
+RUN pip install --quiet --upgrade pip setuptools wheel &&  \
+    pip install -e .
 
-##########################################################
-# SMAC image
-FROM jax-core AS sc2
-## Install smac environment
-RUN apt-get -y install git
-RUN pip install .[sc2]
-# We use the pz wrapper for smac
-RUN pip install .[pz]
-ENV SC2PATH /home/app/mava/3rdparty/StarCraftII
-##########################################################
+# Need to use specific cuda versions for jax
+ARG USE_CUDA=true
+RUN if [ "$USE_CUDA" = true ] ; \
+    then pip install "jax[cuda11_pip]<=0.4.13" -f "https://storage.googleapis.com/jax-releases/jax_cuda_releases.html" ; \
+    fi
```

Comment:
To be able to run on a TPU we would actually need to have a completely separate docker file since running on the TPU won't work with the CUDA base image we are using here. A possible solution would be to have a `Dockerfile.gpu` and `Dockerfile.tpu` but not very many people have got TPU access. So I think it is sufficient to only support GPUs for now.

Reply:
Great point